src/components/shared/Map/index.tsx

```diff
@@ -1,35 +1,57 @@
 import 'leaflet/dist/leaflet.css';
+import { useEffect, useState } from 'react';
 import { MapContainer, TileLayer } from 'react-leaflet';
 
+import getRecent from '@/services/DataService';
+import { getLocation, getLocations } from '@/services/StationsService';
+import { type SensorData } from '@/types/models/SensorData';
+import {
+  type SensorLocation,
+  type SensorLocationData
+} from '@/types/models/SensorLocation';
+
 import MapPointer from './MapPointer';
 
 export default function Map() {
+  const [locations, setLocations] = useState<[SensorLocation]>([]);
+  const [sensorData, setSensorData] = useState<[SensorData]>([]);
+  useEffect(() => {
+    void getLocations().then(async res => {
+      setLocations(res);
+      const data = await getRecent();
+      setSensorData(data);
+    });
+    return () => {};
+  }, []);
+
+  let i = 0;
+  const sensorLocationData: [SensorLocationData] = [];
+  sensorData.forEach((value: SensorData) => {
+    const sensorLocation: SensorLocation = getLocation(value.code, locations);
+    sensorLocationData.push({
+      data: value,
+      location: sensorLocation,
+      key: i
+    });
+    i += 1;
+  });
   return (
     <MapContainer center={[50.05, 19.93]} zoom={12.5} className='w-full h-full'>
       <TileLayer url='https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png' />
-      <MapPointer
-        center={[50.05, 19.93]}
-        radius={20}
-        text='21'
-        sensorData={{
-          date: new Date(),
-          name: 'Kraków, ul. Wrocławska 20',
-          data: [
-            {
-              name: 'PM1',
-              value: 21
-            },
-            {
-              name: 'PM2.5',
-              value: 21
-            },
-            {
-              name: 'PM10',
-              value: 0.003
-            }
-          ]
-        }}
-      />
+      {sensorLocationData.length > 0
+        ? sensorLocationData.map(value =>
+            value.location.n && value.location.e ? (
+              <MapPointer
+                key={value.key}
+                center={[value.location.n, value.location.e]}
+                radius={20}
+                text='21'
+                color='#c90e0e'
+                sensorData={value.data}
+              />
+            ) : null
+          )
+        : null}
```

Comment:
```suggestion
      {sensorLocationData.map(value => (
        <MapPointer
          key={value.key}
          center={[value.location.n, value.location.e]}
          radius={20}
          text='21'
          color='#c90e0e'
          sensorData={value.data}
        />
      ))}
```
The first ternary expression is not necessary as if the `sensorLocationData` is empty, you will get an empty array after calling `map` and the result will be the same. So you don't have to check if an array is empty by yourself.

The second check is also not necessary as the `location` type suggests that both `n` and `e` properties will always be numbers and aren't expected to be nulls.